### PR TITLE
ref(profiling) implement edge detection on profiling tooltip

### PR DESCRIPTION
**Pull request by JonasBa**
Implement edge detection on profiling tooltip. 

We used to have no edge detection, meaning that tooltips could overflow the screen on x axis and become partially hidden. This change implements edge detection, ensuring the tooltip is always contained inside the container (body)

Video shows new vs previous behavior

https://github.com/user-attachments/assets/46011a52-92c6-4c77-9883-07769be8b13c

### CODE REVIEW

File: static/app/components/profiling/boundTooltip.tsx [hunk 1]
```diff
@@ -6,39 +6,78 @@ import {space} from 'sentry/styles/space';
...
+    left = cursor[0] - tooltip.width;
+  } else {
+    // when the tooltip is on the left, we need to add the padding from left side of the container
+    left += CURSOR_LEFT_OFFSET_PX;
+  }
+
+  if (left <= -canvas.left + WIDTH_OFFSET) {
+    // when the tooltip is on the left, we need to add the padding from left side of the container
+    left = -canvas.left + WIDTH_OFFSET;
+  }
```
Comment (Zylphrex) [inline static/app/components/profiling/boundTooltip.tsx:56]: Why is it `-canvas.left`?

File: static/app/components/profiling/boundTooltip.tsx [hunk 2]
```diff
@@ -6,39 +6,78 @@ import {space} from 'sentry/styles/space';
...
 
-  if (cursor[0] > container.width / 2) {
-    left = cursor[0] - tooltip.width; // No offset is applied here as tooltip is placed to the left
+  if (left + canvas.x + tooltip.width >= container.x + container.width - WIDTH_OFFSET) {
+    const offset = left + canvas.x + tooltip.width - (container.x + container.width);
+    // when the tooltip is on the right, we need to subtract the padding from right side of the container
+    left -= offset + WIDTH_OFFSET;
```
Comment (Zylphrex) [inline static/app/components/profiling/boundTooltip.tsx:61]: Why is it `canvas.x`?

File: static/app/components/profiling/boundTooltip.tsx [hunk 3]
```diff
@@ -6,39 +6,76 @@ import {space} from 'sentry/styles/space';
...
+
+  // Cursor is relative to canvas, not container
+  const cursorRelativeToContainer = cursorLeft + canvas.x;
+
+  let left = cursorLeft;
+
+  left =
+    cursorRelativeToContainer > container.width / 2
+      ? cursorLeft - tooltip.width
+      : (left += CURSOR_LEFT_OFFSET_PX);
```
Comment (Zylphrex) [inline static/app/components/profiling/boundTooltip.tsx:52]:
```suggestion
  let left =
    cursorRelativeToContainer > container.width / 2
      ? cursorLeft - tooltip.width
      : cursorLeft + CURSOR_LEFT_OFFSET_PX;
```

File: static/app/components/profiling/boundTooltip.tsx [hunk 4]
```diff
@@ -6,39 +6,76 @@ import {space} from 'sentry/styles/space';
...
-  if (cursor[0] > container.width / 2) {
-    left = cursor[0] - tooltip.width; // No offset is applied here as tooltip is placed to the left
+  if (left + canvas.left - WIDTH_OFFSET <= 0) {
+    left = -canvas.left + WIDTH_OFFSET;
+  } else if (right >= container.width - WIDTH_OFFSET) {
+    left = container.width - tooltip.width - canvas.left - WIDTH_OFFSET;
   }
 
-  return `translate(${left || 0}px, ${top || 0}px)`;
+  return `translate(${left}px, ${cursorTop + CURSOR_TOP_OFFSET_PX}px)`;
```
Comment (Zylphrex) [inline static/app/components/profiling/boundTooltip.tsx:62]: A comment explaining the final `left` is relative to the canvas would be helpful here.

**Review by Zylphrex — APPROVED** [review]

File: static/app/components/profiling/boundTooltip.tsx (hunk 4 above)
Comment (JonasBa) [reply static/app/components/profiling/boundTooltip.tsx:60]: Yeah, I added this where we perform the bound checks

**Pull request merged**